### PR TITLE
flamenco: update to cleaner cluster logic

### PR DESCRIPTION
**Pull request by kbhargava-jump**

### CODE REVIEW

File: contrib/test/test_firedancer.sh [hunk 1]
```diff
@@ -142,7 +142,7 @@ name = \"fd1test\"
         funk_sz_gb = 32
         funk_rec_max = 10000000
         funk_txn_max = 1024
-        cluster_version = 1180
+        cluster_version =  1.18.0
```
Comment (riptl) [inline contrib/test/test_firedancer.sh:145]: This is parsed as a string, so it should be wrapped in quotes.

File: src/app/fdctl/config.h [hunk 2]
```diff
@@ -269,7 +269,7 @@ typedef struct {
       char  snapshot[ PATH_MAX ];
       char  status_cache[ PATH_MAX ];
       ulong tpool_thread_count;
-      uint  cluster_version;
+      char  cluster_version[ PATH_MAX ];
```
Comment (riptl) [inline src/app/fdctl/config.h:272]: PATH_MAX is overkill, you can use 32 or so here. The TOML parser automatically picks up the max string length.

File: src/app/ledger/main.c [hunk 3]
```diff
@@ -1442,7 +1445,7 @@ initial_setup( int argc, char ** argv, fd_ledger_args_t * args ) {
   args->checkpt_path            = checkpt_path;
   args->checkpt_freq            = checkpt_freq;
   args->checkpt_mismatch        = checkpt_mismatch;
-  args->cluster_version         = cluster_version;
+  sscanf(cluster_version, "%u.%u.%u", &args->cluster_version[0], &args->cluster_version[1], &args->cluster_version[2]);
```
Comment (riptl) [inline src/app/ledger/main.c:1448]: Incorrect formatting and missing error check if sscanf fails

**Review by riptl — COMMENTED** [review]
Good change. Please address above before merging

**Review by jumpsiegel — DISMISSED** [review]

**Review by ravyu-jump — APPROVED** [review]

**Pull request merged**